crates/sui-types/src/sui_system_state/mod.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::base_types::{AuthorityName, SuiAddress};
+use crate::committee::{CommitteeWithNetAddresses, EpochId, ProtocolVersion};
+use crate::dynamic_field::{derive_dynamic_field_id, Field};
+use crate::error::SuiError;
+use crate::storage::ObjectStore;
+use crate::{id::UID, SUI_FRAMEWORK_ADDRESS, SUI_SYSTEM_STATE_OBJECT_ID};
+use anemo::PeerId;
+use anyhow::Result;
+use enum_dispatch::enum_dispatch;
+use move_core_types::language_storage::TypeTag;
+use move_core_types::value::MoveTypeLayout;
+use move_core_types::{ident_str, identifier::IdentStr, language_storage::StructTag};
+use move_vm_types::values::Value;
+use multiaddr::Multiaddr;
+use narwhal_config::{Committee as NarwhalCommittee, WorkerCache};
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeMap, HashMap};
+use tracing::error;
+
+use self::sui_system_state_inner_v1::{SuiSystemStateInnerV1, ValidatorMetadata};
+
+pub mod sui_system_state_inner_v1;
+
+const SUI_SYSTEM_STATE_WRAPPER_STRUCT_NAME: &IdentStr = ident_str!("SuiSystemState");
+
+pub const SUI_SYSTEM_MODULE_NAME: &IdentStr = ident_str!("sui_system");
+pub const ADVANCE_EPOCH_FUNCTION_NAME: &IdentStr = ident_str!("advance_epoch");
+pub const ADVANCE_EPOCH_SAFE_MODE_FUNCTION_NAME: &IdentStr = ident_str!("advance_epoch_safe_mode");
+pub const CONSENSUS_COMMIT_PROLOGUE_FUNCTION_NAME: &IdentStr =
+    ident_str!("consensus_commit_prologue");
+
+pub const INIT_SYSTEM_STATE_VERSION: u64 = 1;
+
+/// Rust version of the Move sui::sui_system::SuiSystemState type
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct SuiSystemStateWrapper {
+    pub id: UID,
+    pub version: u64,
+}
+
+impl SuiSystemStateWrapper {
```

Comment:
I find it slightly confusing the relationship between `SuiSystemStateWrapper`, `SuiSystemState` enum and `SuiSystemStateTrait`, and when each should be used. Can we have a few lines of comments explaining it?